spring-amqp/src/main/java/org/springframework/amqp/support/postprocessor/AbstractCompressingPostProcessor.java

```diff
@@ -74,13 +78,19 @@ public Message postProcessMessage(Message message) throws AmqpException {
 		try {
 			OutputStream zipper = getCompressorStream(zipped);
 			FileCopyUtils.copy(new ByteArrayInputStream(message.getBody()), zipper);
-			MessageProperties messageProperties = message.getMessageProperties();
-			String currentEncoding = messageProperties.getContentEncoding();
-			messageProperties
-					.setContentEncoding(getEncoding() + (currentEncoding == null ? "" : ":" + currentEncoding));
+			MessageProperties originalProperties = message.getMessageProperties();
+			MessageBuilderSupport<MessageProperties> messagePropertiesMessageBuilder =
+					MessagePropertiesBuilder.fromClonedProperties(originalProperties)
```

Comment:
It's not at all clear why this change is needed. Seems like unnecessary overhead.

Reply:
We build here a new message and modify existing properties new new content encoding like `gzip:`.
So, the original message gets wrong properties. 
We don't here on the producer side a guarantee that the provided message is not built by end-user.
In fact the `AsyncRabbitTemplateTests` uses the same `fooMessage` for different sends and I only figured out in debug that it fails because of `gzip:gzip:` prefix in the content encoding for the second send.
It is OK on the consumer side though, since message is created by the Framework and we really don't propagate it to the target end-user code.

Does it make sense?

Reply:
Got it; thanks, but perhaps it should be an option (default dont clone,)? In most cases, users won't keep the original message around - especially if `convertAndSend()` is used.